src/plugins/ui_feature/datavalidation_insertion.ts

```diff
@@ -0,0 +1,35 @@
+import { getCellPositionsInRanges, isBoolean } from "../../helpers";
+import { CellValueType, Command, isMatrix } from "../../types/index";
+import { UIPlugin } from "../ui_plugin";
+
+export class DataValidationInsertionPlugin extends UIPlugin {
+  handle(cmd: Command) {
+    switch (cmd.type) {
+      case "ADD_DATA_VALIDATION_RULE":
+        if (cmd.rule.criterion.type === "isBoolean") {
+          const ranges = cmd.ranges.map((range) => this.getters.getRangeFromRangeData(range));
+          for (const position of getCellPositionsInRanges(ranges)) {
+            const cell = this.getters.getCell(position);
+            const evaluatedCell = this.getters.getEvaluatedCell(position);
+
+            if (!cell?.content) {
+              this.dispatch("UPDATE_CELL", { ...position, content: "FALSE" });
+            } else if (cell?.content && evaluatedCell.type === CellValueType.empty) {
```

Comment:
Actuall, if I comment the entire body of this `if` all test pass. What we actually want is just to avoid the dispatch in this case, right ?